Jenkinsfile

```diff
@@ -1,29 +1,39 @@
 elifePipeline {
     def commit
+    stage 'Checkout', {
+        checkout scm
+        commit = elifeGitRevision()
+    }
+
     elifeOnNode(
         {
-            stage 'Checkout', {
+            stage 'Build images', {
                 checkout scm
-                commit = elifeGitRevision()
+                sh 'docker-compose -f docker-compose.ci.yml build'
             }
 
-            stage 'Container image', {
-                sh 'docker-compose -f docker-compose.ci.yml build'
-                sh 'chmod 777 build/ && docker-compose -f docker-compose.ci.yml run ci ./project_tests.sh'
-                step([$class: "JUnitResultArchiver", testResults: 'build/phpunit.xml'])
-                sh 'docker-compose -f docker-compose.ci.yml run ci ./smoke_tests.sh web'
+            stage 'Project tests', {
+                try {
+                    sh 'chmod 777 build/ && docker-compose -f docker-compose.ci.yml run --rm ci ./project_tests.sh'
+                    step([$class: "JUnitResultArchiver", testResults: 'build/phpunit.xml'])
+                    sh 'docker-compose -f docker-compose.ci.yml run --rm ci ./smoke_tests.sh web'
+                } finally {
+                    sh 'docker-compose -f docker-compose.ci.yml stop'
+                }
+            }
+
+            elifeMainlineOnly {
+                stage 'Push images', {
```

Comment:
they end up at:
- https://hub.docker.com/r/elifesciences/annotations_cli/tags/
- https://hub.docker.com/r/elifesciences/annotations_fpm/tags/

Reply:
The idea is downstream servers will be able to pull the images without building them from source, shaving off minutes from every pipeline run. At the moment we are still targeting EC2 instances running only one application, but once the image-based deployment works we can do more.
Of course we only want to do this for passing builds of `develop` (`elifeMainlineOnly`). See also https://github.com/elifesciences/elife-libraries-formula/pull/5 for the credentials to do this.